p2p/src/sync/peer.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): A. Altonen
+#![allow(unused)]
+
+use crate::{
+    error::{self, P2pError},
+    event,
+    net::{self, NetworkService},
+    sync,
+};
+use common::chain::block::{Block, BlockHeader};
+use logging::log;
+use std::collections::{HashSet, VecDeque};
+use tokio::sync::mpsc;
+
+/// State of the peer
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum PeerSyncState {
+    /// Peer state is unknown
+    Unknown,
+
+    /// Peer is uploading blocks to local node
+    UploadingBlocks(BlockHeader),
+
+    /// Peer is uploading headers to local node
+    UploadingHeaders,
+
+    /// Peer is idling and can be used for block requests
+    Idle,
+}
+
+/// Syncing-related context of the peer
+pub struct PeerContext<T>
+where
+    T: NetworkService,
+{
+    /// Unique peer ID
+    peer_id: T::PeerId,
+
+    /// State of the peer
+    state: PeerSyncState,
+
+    /// List of headers the peer must download
+    work: VecDeque<BlockHeader>,
```

Comment:
It's not quite clear to me what does this track. Is it the list of headers for local node to download or does it somehow track remote peers state?

Reply:
It tracks the blocks the local node still needs to download from the remote peer. I've rewritten the comment to explain the purpose more clearly